salt/modules/cmdmod.py

```diff
@@ -100,25 +101,20 @@ def _check_cb(cb_):
     return lambda x: x
 
 
-def _python_shell_default(python_shell, shell=False):
+def _python_shell_default(python_shell, __pub_jid):
     """
     Set python_shell default based on the shell parameter and __opts__['cmd_safe']
     """
-    if shell:
-        if salt.utils.platform.is_windows():
-            # On Windows python_shell / subprocess 'shell' parameter must always be
-            # False as we prepend the shell manually
-            return False
-        else:
-            # Non-Windows requires python_shell to be enabled
-            return True if python_shell is None else python_shell
-    else:
-        try:
-            if __opts__.get("cmd_safe", True) is False and python_shell is None:
-                # Override-switch for python_shell
-                return True
-        except NameError:
-            pass
+    try:
+        # Default to python_shell=True when run directly from remote execution
+        # system. Cross-module calls won't have a jid.
+        if __pub_jid and python_shell is None:
+            return True
+        elif __opts__.get("cmd_safe", True) is False and python_shell is None:
+            # Override-switch for python_shell
+            return True
```

Comment:
Are there any tests for cmd_safe? I did a quick search and did not notice any.

Reply:
There were none before it was removed in https://github.com/saltstack/salt/pull/68156